### ISSUE TITLE
Consider adding an Advanced tab to allow people to manually select certain transformations

### ISSUE THREAD

**Issue by euan-forrester** [euan-forrester/save-file-converter#167]
For example:
- Resizing a save by adding/removing padding (allow user to select `0x00` or `0xFF` padding).
  - Should we have a dropdown for sizes, because people may have trouble typing a power-of-2 size?
  - It may also be confusing to find the "correct" size vs the "size on disc" that the OS may show them.
  - Make the list be the superset of all sizes in `PlatformSaveSizes.js`?
  - Maybe we should display the current size of the file as well
  - Resizing should probably be the first tab as well, since it's the most common thing that comes up when people are asking for support
- Endian swapping (allow the user to select 16, 32, 64 byte endian swapping)
- Byte expanding/contracting Genesis files (allow user to select byte expansion byte: `0x00`, `0xFF`, or double byte e.g. `0xA` -> `0xAA`)
- Remove a header/footer (allow user to specify number of bytes, or attempt to autodetect based on making the file a size that's a power of 2)
- Add a header/footer by copying it from another save file
- Allow people to select a certain range of offsets to slice out, for example to allow: https://www.reddit.com/r/AnaloguePocket/comments/10ampwi/solution_for_extracting_pokemon_emerald_cartridge/
- Maybe decompressing the data using a common compression scheme like `.zip` or `.lzo`
- File comparison: files are exactly the same, or not

See here for ideas: https://forums.launchbox-app.com/topic/66417-n64-save-converter-tools/

**Comment by euan-forrester** [comment]
We should also have instructions somewhere (maybe the FAQ on Discord?) indicating to try first removing a header then truncating the file to be the same size as a known good file. I suspect this will handle many cases.

**Comment by omardelarosa** [comment]
+1 on adding this:

> Maybe decompressing the data using a common compression scheme like .zip or .lzo

I actually ran into a gnarly issue where the .srm files I was trying to convert had been compressed by RetroArch and the output .sav files were not decompressed correctly.  It took me a while to debug the issue, had to use a hex editor to figure out what was going and see the `.rz` / `rzip` header and then re-create the .srms with compression turned off.

tl;dr - having advanced options like (de-)compression in the tool would save a ton of grief.

**Comment by euan-forrester** [comment]
Nice! That's great to hear that it would be useful. I'm looking forward to adding this feature!

**Comment by euan-forrester** [comment]
I've started looking at this more closely, and apparently the format is not `rzip` but actually `zlib`. "rzip" in this case apparently stands for "Retroarch zip" and not the actual `rzip` tool: https://github.com/libretro/RetroArch/issues/14031#issuecomment-1159400581

**Comment by omardelarosa** [comment]
Oh good to know @euan-forrester !  But yeah, that sounds about right because the context I was using this all in was a Recalbox btw.

**Comment by euan-forrester** [comment]
@omardelarosa You can find this now at https://savefileconverter.com/#/utilities/advanced?tab=compression

**Comment by omardelarosa** [comment]
Amazing! Thanks so much for adding this! Will be really helpful for moving
my saves around.

**Comment by euan-forrester** [comment]
No problem! Please let me know how you get on with it
